Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -825,8 +830,10 @@ void DataProcessingDevice::handleData(DataProcessorContext& context, FairMQParts
           auto headerIndex = 2 * pi;
           auto payloadIndex = 2 * pi + 1;
           assert(payloadIndex < parts.Size());
+          auto dh = o2::header::get<DataHeader*>(parts.At(headerIndex)->GetData());
           auto relayed = relayer.relay(std::move(parts.At(headerIndex)),
-                                       std::move(parts.At(payloadIndex)));
+                                       &parts.At(payloadIndex), dh->splitPayloadParts * 2 - 1);
```

Comment:
@matthiasrichter I think I was lacking a `*2` here...

Reply:
yes, now the logic is consistent